lib/Conversion/ImportVerilog/Expressions.cpp

```diff
@@ -429,10 +429,22 @@ struct ExprVisitor {
   }
 
   Value visit(const slang::ast::MemberAccessExpression &expr) {
-    return builder.create<moore::StructExtractOp>(
-        loc, context.convertType(*expr.type),
-        builder.getStringAttr(expr.member.name),
-        context.convertExpression(expr.value()));
+    auto type = expr.value().type;
+    if (type->isStruct()) {
+      return builder.create<moore::StructExtractOp>(
+          loc, context.convertType(*expr.type),
+          builder.getStringAttr(expr.member.name),
+          context.convertExpression(expr.value()));
+    }
+    if (type->isPackedUnion() || type->isUnpackedUnion()) {
+      return builder.create<moore::UnionExtractOp>(
+          loc, context.convertType(*expr.type),
+          builder.getStringAttr(expr.member.name),
+          context.convertExpression(expr.value()));
+    }
+    mlir::emitError(loc, "unsupported symbol kind: ")
+        << slang::ast::toString(type->kind);
+    return {};
```

Comment:
nit: Could you add a related case for this error?

Reply:
Sorry, I can not imagine for this, but error message should be printed if other unexpected symbol kinds are received. So I recommand remain this message rather than print nothing for future debugging.